messaging-gateway/messaging-gateway.bal

```diff
@@ -0,0 +1,54 @@
+import ballerina/http;
+import ballerina/io;
+
+type ActivateLine readonly & record {|
+    string customerId;
+    string phoneNumber;
+    "NEW ACTIVATION"|"TRANSFER" activationType;
+    string planCode;
+    string activationDate;
+|};
+
+type ActivationResponse readonly & record {|
+    string phoneNumber;
+    "REJECTED"|"APPROVED" status;
+|};
+
+type ActivationSuccess readonly & record {|
+    string phoneNumber;
+|};
+
+type ActivationFailure readonly & record {|
+    string phoneNumber;
+|};
+
+public class LineConnectionManager {
+
+    private final http:Client lineClinet;
+
+    public function init() returns error? {
+        self.lineClinet = check new ("http://api.telcox.com.balmock.io");
+    }
+
+    public function activateLine(ActivateLine activateLine) returns ActivationSuccess|ActivationFailure {
+        ActivationResponse|error activationResponse = self.lineClinet->/plans/[activateLine.planCode]/activate.post(activateLine);
+        if activationResponse is error || activationResponse.status == "REJECTED" {
+            return <ActivationFailure>{phoneNumber: activateLine.phoneNumber};
+        }
+        return <ActivationSuccess>{phoneNumber: activateLine.phoneNumber};
```

Comment:
Ballerina has a structural type system. Both records are structurally similar. So we can't tell them apart. need to change.

Reply:
Updated.